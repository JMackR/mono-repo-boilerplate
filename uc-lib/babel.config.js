module.exports = {
  presets: [["@babel/preset-env", { targets: { node: "current" } }], "module:metro-react-native-babel-preset"],
  plugins: ["@babel/plugin-proposal-class-properties", "inline-json-import"],
  sourceType: "unambiguous",
}
