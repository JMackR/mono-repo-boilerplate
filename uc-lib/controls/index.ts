export * from "./image"
export * from "./button"
export * from "./text"
export * from "./activity-indicator"
export * from "./separator"
export * from "./list"
export * from "./section-list"
export * from "./badge"
export * from "./toggle"
export * from "./login-button"
export * from "./layout"
export * from "./background"
export * from "./grid"
export * from "./touchable"
export * from "./view-block"
export * from "./text-entry"
export * from "./scroll-view"
export * from "./circle"
export * from "./slider"
export * from "./clickable"
export * from "./map"
export * from "./visibility-check"
export * from "./strike-through-text"
// export * from "./stripe-text-entry"
export * from "./input"
