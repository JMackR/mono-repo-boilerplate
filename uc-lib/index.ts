export * from "./animations"
export * from "./assets"
export * from "./controls"
export * from "./themes"
export * from "./widgets"
export * from "./utilities"
export * from "./hooks"
export * from "./validators"
