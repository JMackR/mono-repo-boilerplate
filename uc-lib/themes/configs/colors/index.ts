export * from "./light-mode"
export * from "./dark-mode"
