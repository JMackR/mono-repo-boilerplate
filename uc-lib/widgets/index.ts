export * from "./account"
export * from "./attribute-list"
export * from "./breadcrumb"
export * from "./dialog/"
export * from "./empty-state"
export * from "./expanding-text-section"
export * from "./flexible-row"
export * from "./full-screen-spinner"
export * from "./inbox-row"
export * from "./indicator-dots"
export * from "./item-preview"
export * from "./listing-photo-list"
export * from "./pill-list"
export * from "./segmented-control"
export * from "./selectable-row"
export * from "./tooltip"
export * from "./user-profile"
export * from "./search"
