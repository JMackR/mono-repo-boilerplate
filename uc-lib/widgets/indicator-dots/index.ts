export * from "./indicator-dots-props"
export * from "./indicator-dots"
