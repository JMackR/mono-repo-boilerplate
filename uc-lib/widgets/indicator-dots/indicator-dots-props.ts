export interface IndicatorDotsProps {
  count: number
  selectedIndex: number
}
