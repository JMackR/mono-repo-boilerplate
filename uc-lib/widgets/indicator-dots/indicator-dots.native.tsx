import React, { FC } from "react"
import { Stack, Flex } from "../../controls"
import { IndicatorDotsProps } from "./indicator-dots-props"
import { View, StyleSheet } from "react-native"
import { useColor } from "uc-lib/themes"

const CHILD_SEPARATION_STEP = 2

const IndicatorDots: FC<IndicatorDotsProps> = props => {
  const { count, selectedIndex } = props

  const dots = []
  for (let i = 0; i < count; i++) {
    dots.push(<IndicatorDot selected={i === selectedIndex} key={i} />)
  }

  return (
    <Flex direction="column" grow={1} axisDistribution="center" crossAxisDistribution="center">
      <Stack direction="row" grow={1} childSeparationStep={CHILD_SEPARATION_STEP}>
        {dots}
      </Stack>
    </Flex>
  )
}

interface IndicatorDotsProps {
  selected: boolean
}

const IndicatorDot: FC<IndicatorDotsProps> = props => {
  const { selected } = props
  const { colors } = useColor()

  const DotStyles = StyleSheet.create({
    dot: {
      borderRadius: 8,
      width: 8,
      height: 8,
      backgroundColor: selected ? colors.green : colors.limestone,
    },
  })

  return <View style={DotStyles.dot} />
}

export { IndicatorDots }
