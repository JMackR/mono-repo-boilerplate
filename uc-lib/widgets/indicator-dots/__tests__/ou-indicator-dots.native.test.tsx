import React from "react"
import { render } from "react-native-testing-library"
import { useColor } from "uc-lib"
import { _LIGHT_MODE_COLOR_THEME as TEST_THEME } from "uc-lib/themes/configs/colors"
import { IndicatorDots } from "../indicator-dots.native"

jest.mock("uc-lib/themes/hooks/use-color")
useColor.mockReturnValue(TEST_THEME)

test("renders one dot with selection", () => {
  const tree = render(<IndicatorDots count={1} selectedIndex={0} />)
  expect(tree).toMatchSnapshot()
})

test("renders three dots with selection", () => {
  const tree = render(<IndicatorDots count={3} selectedIndex={1} />)
  expect(tree).toMatchSnapshot()
})
