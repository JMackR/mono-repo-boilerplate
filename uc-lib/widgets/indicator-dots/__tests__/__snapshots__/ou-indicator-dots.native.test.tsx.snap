// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders one dot with selection 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "flexWrap": undefined,
      "height": undefined,
      "justifyContent": "center",
      "overflow": undefined,
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "overflow": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#00A87E",
          "borderRadius": 8,
          "height": 8,
          "width": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`renders three dots with selection 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "flexWrap": undefined,
      "height": undefined,
      "justifyContent": "center",
      "overflow": undefined,
      "width": undefined,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": undefined,
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": undefined,
        "justifyContent": "flex-start",
        "overflow": undefined,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E2E2E2",
          "borderRadius": 8,
          "height": 8,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "height": undefined,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#00A87E",
          "borderRadius": 8,
          "height": 8,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "height": undefined,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#E2E2E2",
          "borderRadius": 8,
          "height": 8,
          "width": 8,
        }
      }
    />
  </View>
</View>
`;
